doc/developer/design/20240919_replica_expiration.md

```diff
@@ -0,0 +1,86 @@
+# Replica Expiration
+
+- Associated issues/prs: [#26029](https://github.com/MaterializeInc/materialize/issues/26029) [#29587](https://github.com/MaterializeInc/materialize/pull/29587)
+
+## The Problem
+
+Temporal filters currently require Materialize to maintain all future retractions
+of data that is currently visible. For long windows, the retractions could be at
+timestamps beyond our next scheduled restart, for example during our weekly
+DB releases.
+
+For instance, in the example below, the temporal filter in the `last_30_sec`
+view causes two diffs to be generated for every row inserted into `events`, the
+row itself and a retraction 30 seconds later. However, if the replica is
+restarted in the next 30 seconds, the retraction diff is never processed, making
+it unnecessary to generate and store that extra diff.
+
+```sql
+-- Create a table of timestamped events.
+CREATE TABLE events (
+content TEXT,
+event_ts TIMESTAMP
+);
+-- Create a view of events from the last 30 seconds.
+CREATE VIEW last_30_sec AS
+SELECT event_ts, content
+FROM events
+WHERE mz_now() <= event_ts + INTERVAL '30s';
+
+INSERT INTO events VALUES ('hello', now());
+
+COPY (SUBSCRIBE (SELECT event_ts, content FROM last_30_sec)) TO STDOUT;
+1686868190714   1       2023-06-15 22:29:50.711 hello  -- now()
+1686868220712   -1      2023-06-15 22:29:50.711 hello  -- now() + 30s
+```
+
+## Success Criteria
+
+Diffs associated with timestamps beyond a set expiration time---mainly the
+retractions generated in temporal filters---are dropped without affecting
+correctness, resulting in lower memory utilization.
```

Comment:
Also CPU utilization, in the case of arrangement maintenance!

Reply:
Updated!